### PR TITLE
Add missed attributes for Parameter layer

### PR DESCRIPTION
**Pull request by achetver**
Root cause analysis: When we use `--input` option MO creates IR without necessary attributes in `Parameter` layer and it leads to error while we load model to IE. It happens because we use `create_node_with_data()` operation to create this layer and manually update map with operation attributes to save using `update_ie_fields` function.

Solution: Update `add_input_op_input_port_with_data()` and `add_input_op_output_port_with_data()` functions to use `create_node_with_data()` instead of `create_node_with_data()` method and new GraphAPI .

Ticket: 55009

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests: Existed tests were updated - removed redundant checks;
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  Model Optimizer IR Reader check: Done manually.

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

**Comment by achetver** [comment]
@iimironov @sadolini @evolosen Please, review these changes.

**Review by iimironov — APPROVED** [review]

**Review by evolosen — APPROVED** [review]

**Comment by achetver** [comment]
@lazarevevgeny Please, review this PR.

File: model-optimizer/mo/front/extractor.py [hunk 1]
```diff
@@ -312,6 +314,8 @@ def update_ie_fields(attrs: dict, ir_version = None):
                         'coeff',
                         ('ratio', lambda node: attr_getter(node, 'ratio')),
                         'size',
+                        ('shape', lambda node: ','.join([str(i) for i in node.shape]) if node.has_valid('shape') else None),
+                        ('element_type', lambda node: np_data_type_to_destination_type(node.data_type) if node.has_valid('data_type') else None),
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/extractor.py:318]: We should not modify this list. It is a legacy approach which we do not rely on anymore. Instead modify the code which cuts the model and create Parameter nodes using Parameter class.

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/mo/front/extractor.py (hunk 1 above)
Comment (achetver) [reply model-optimizer/mo/front/extractor.py]: Ok, done.

**Comment by achetver** [comment]
@iimironov @evolosen @sadolini Please, review this PR again, because the main logic was changed.

**Review by evolosen — APPROVED** [review]

File: model-optimizer/mo/front/extractor.py [hunk 2]
```diff
@@ -805,10 +805,14 @@ def add_input_op_input_port_without_data(graph: Graph, node_id: str, input_op, e
 
 
 def add_input_op_input_port_with_data(graph: Graph, node_id: str, input_op, edge_attrs: dict):
-    input_data_node = input_op.create_node_with_data()
-    input_node = input_data_node.in_node()
-    graph.add_edge(input_data_node.id, node_id, **edge_attrs)
-    update_ie_fields(graph.node[input_node.id])
+    input_node = input_op.create_node(edge_attrs=edge_attrs)
```
Comment (sadolini) [inline model-optimizer/mo/front/extractor.py:808]: Check that graph is not on front stage, so data nodes will be really created

**Review by sadolini — APPROVED** [review]

File: model-optimizer/mo/front/extractor.py (hunk 2 above)
Comment (achetver) [reply model-optimizer/mo/front/extractor.py:808]: Both updated functions called for graph after shape inference, but I can such assert.

Comment (achetver) [reply model-optimizer/mo/front/extractor.py:810]: Done.

File: model-optimizer/mo/front/extractor.py [hunk 3]
```diff
@@ -805,10 +805,16 @@ def add_input_op_input_port_without_data(graph: Graph, node_id: str, input_op, e
 
 
 def add_input_op_input_port_with_data(graph: Graph, node_id: str, input_op, edge_attrs: dict):
-    input_data_node = input_op.create_node_with_data()
-    input_node = input_data_node.in_node()
-    graph.add_edge(input_data_node.id, node_id, **edge_attrs)
-    update_ie_fields(graph.node[input_node.id])
+    assert graph.stage == 'middle', 'add_input_op_input_port_with_data() function can be used only for graph after ' \
```
Comment (lazarevevgeny) [inline model-optimizer/mo/front/extractor.py:808]: @jane-intel , could you take a look at these changes? Do we expect to keep some attributes from the data node? I see that in the modified unit test we removed `assert` checking existence of data nodes. I guess that these asserts exist for some reason.

**Review by iimironov — APPROVED** [review]

File: model-optimizer/mo/front/extractor.py (hunk 3 above)
Comment (achetver) [reply model-optimizer/mo/front/extractor.py:808]: I checked - updated function saves necessary attributes. Related checks was added in test.

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**